File: app/schemas/evidence.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class EvidenceOut(BaseModel):
    id: UUID
    material_entry_id: UUID
    file_name: str
    file_type: str
    content_type: str
    file_size: int
    file_hash: str
    evidence_type: str
    duplicate_flag: bool
    storage_path: str
    uploaded_by: UUID
    uploaded_at: datetime

    class Config:
        orm_mode = True


class EvidenceListOut(BaseModel):
    total: int
    items: list[EvidenceOut]

File: frontend/src/types/evidence.ts
export type Evidence = {
  id: string;
  material_entry_id: string;
  file_name: string;
  file_type: string;
  content_type: string;
  file_size: number;
  file_hash: string;
  evidence_type: string;
  duplicate_flag: boolean;
  storage_path: string;
  uploaded_by: string;
  uploaded_at: string;
};
